tests/PhpBrew/Patches/Apache2ModuleNamePatchTest.php

```diff
@@ -12,34 +12,47 @@
  */
 class Apache2ModuleNamePatchTest extends PatchTestCase
 {
-    public function testPatch()
+    public function testPatchForPHP55()
+    {
+        $this->targetPatchVersion('5.5.17');
+    }
+
+    public function testPatchForPHP74()
+    {
+        $this->targetPatchVersion('7.4.0');
+    }
+
+    public function targetPatchVersion($targetVersion)
     {
         $logger = new Logger();
         $logger->setQuiet();
 
-        $fromVersion = '5.5.17';
-        $sourceFixtureDirectory = getenv('PHPBREW_FIXTURES_PHP_DIR') . DIRECTORY_SEPARATOR . $fromVersion;
         $sourceDirectory = getenv('PHPBREW_BUILD_PHP_DIR');
 
         if (!is_dir($sourceDirectory)) {
             $this->markTestSkipped("$sourceDirectory does not exist.");
         }
 
-        $this->setupBuildDirectory($fromVersion);
+        $this->setupBuildDirectory($targetVersion);
 
-        $build = new Build($fromVersion);
+        $build = new Build($targetVersion);
         $build->setSourceDirectory($sourceDirectory);
         $build->enableVariant('apxs2');
         $this->assertTrue($build->hasVariant('apxs2'), 'apxs2 enabled');
 
-        $patch = new Apache2ModuleNamePatch();
+        $patch = new Apache2ModuleNamePatch($targetVersion);
         $matched = $patch->match($build, $logger);
         $this->assertTrue($matched, 'patch matched');
         $patchedCount = $patch->apply($build, $logger);
-        $this->assertEquals(107, $patchedCount);
 
-        $sourceExpectedDirectory = getenv('PHPBREW_EXPECTED_PHP_DIR') . DIRECTORY_SEPARATOR . '5.5.17-apxs-patch';
-        $this->assertFileEquals($sourceExpectedDirectory . '/Makefile.global', $sourceDirectory . '/Makefile.global');
+        $sourceExpectedDirectory = getenv('PHPBREW_EXPECTED_PHP_DIR') . DIRECTORY_SEPARATOR . $targetVersion.'-apxs-patch';
+        if (version_compare($targetVersion, '7.4') >= 0) {
```

Comment:
Please avoid conditionals in tests. Instead, you can pass the expected patched count as a parameter together with the version. Also, you can replace the newly introduced test methods with a `@dataProvider`.

Reply:
Done. Use @dataProvider as you requested, instead of conditionals in tests.